lib/validation.js

```diff
@@ -233,7 +233,19 @@ module.exports = function validation (yargs, usage, y18n) {
       if (Array.isArray(value)) {
         value.forEach((i) => self.implies(key, i))
       } else {
-        implied[key].push(value)
+        if (typeof value === 'object') {
+          const valueCopy = Object.assign({}, value)
+
+          for (let k in valueCopy) {
```

Comment:
`for..in` is unsafe, because it can iterate over inherited keys.
You should check if object `hasOwnProperty` like shown here: https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Statements/for...in in `Iterating over own properties only` section or you can use `Object.keys(valueCopy).forEach(function (key)  {... })`